src/components/Filters/index.tsx

```diff
@@ -28,229 +29,344 @@ const FilterWrapper = styled(Box)`
-			<Flex justifyContent="space-between">
-				{shouldRenderComponent('add', renderMode) && (
-					<Button
-						mr={30}
-						disabled={disabled}
-						primary
-						icon={<FontAwesomeIcon icon={faFilter} />}
-						onClick={() => {
-							setshowModal(true);
-							setEditingFilter(null);
-						}}
-						{...addFilterButtonProps}
-						label={shouldHideLabel ? undefined : 'Add filter'}
-					></Button>
-				)}
-
-				{shouldRenderComponent('search', renderMode) && (
-					<SearchWrapper>
-						<Search
-							dark={dark}
-							disabled={disabled}
-							value={searchString}
-							onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
-								setSimpleSearch(e.target.value)
+	public setSimpleSearch(term: string) {
+		this.setState(
+			prevState => {
+				const newFilters = term
+					? SchemaSieve.upsertFullTextSearch(
+							this.state.schema,
+							prevState.filters,
+							term,
+					  )
+					: SchemaSieve.removeFullTextSearch(prevState.filters);
+
+				return {
+					searchString: term,
+					filters: newFilters,
+				};
+			},
+			() => this.emitFilterUpdate(),
+		);
+	}
+
+	public shouldRenderComponent(mode: FilterRenderMode): boolean {
+		// If a render mode is not specified, render all components
+		if (!this.props.renderMode) {
+			return true;
+		}
+
+		const allowedModes = castArray(this.props.renderMode);
+
+		if (includes(allowedModes, 'all')) {
+			return true;
+		}
+
+		return includes(allowedModes, mode);
+	}
+
+	public render() {
+		const { filters } = this.state;
+
+		return (
+			<FilterWrapper mb={3}>
+				<Flex justifyContent="space-between">
+					{this.shouldRenderComponent('add') && (
+						<Button
+							mr={30}
+							disabled={this.props.disabled}
+							primary
+							icon={<FontAwesomeIcon icon={faFilter} />}
+							onClick={() =>
+								this.setState({ showModal: true, editingFilter: null })
+							}
+							label="Add filter"
+							compact={this.props.compact || [false]}
+							{...this.props.addFilterButtonProps}
+						></Button>
+					)}
+
+					{this.shouldRenderComponent('search') && (
+						<SearchWrapper>
+							<Search
+								dark={this.props.dark}
+								disabled={this.props.disabled}
+								value={this.state.searchString}
+								onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+									this.setSimpleSearch(e.target.value)
+								}
+							/>
+						</SearchWrapper>
+					)}
+
+					{this.shouldRenderComponent('views') && (
+						<ViewsMenu
+							dark={this.props.dark}
+							buttonProps={this.props.viewsMenuButtonProps}
+							disabled={this.props.disabled}
+							views={this.state.views || []}
+							schema={this.props.schema}
+							hasMultipleScopes={
+								this.props.viewScopes && this.props.viewScopes.length > 1
 							}
+							setFilters={filters => this.setFilters(filters)}
+							deleteView={view => this.deleteView(view)}
+							renderMode={this.props.renderMode}
+							compact={this.props.compact || [false]}
```

Comment:
```suggestion
							compact={this.props.compact}
```
Since every sub-component should initialize its own default values.